README.md

```diff
@@ -1,26 +1,28 @@
-# Repository template
-A template enabled repository, including all necesary files to open source.
-(create an issue with the following content if you want to track the repo configuration)
+# Band Oracle Contract
+A smart contract for enabling the Band Protocol relay system to keep crypto and fiat
+currency prices updated #onFlow
```

Comment:
So interestingly while the current focus is on trading pair data feeds from the oracle, in future one could send any arbitrary key/value pairs of data through an oracle. The main criteria determining whether it makes sense to use an oracle to provide a given type of data is that it's:

* Time sensitive, thereby warranting low latency delivery
* Data provenance needs to be authoritatively proved for consumers by validators
* Data is temporal in nature

Given that, in the long term one could imagine oracles feeding much more than just pricing data. One example I came up with was regional/global emergency alerts. Doesn't have financial value but sure is valuable to people and meets the other above criteria. 

I'm not saying we design the interfaces to this to be general purpose but at the same time I would want to ensure there's flexibility to get to these other use-cases in future and that the whole system isn't so totally tied to trading pairs. Maybe we solve with a whole new contract, not sure. 

```suggestion
The Cadence smart contract(s) enabling Flow native integration of the Band Protocol oracle network. The oracle network will provide price updates for crypto, fiat and commodities financial data for consumption by Flow applications. 

To learn more about Band Protocol please refer to: https://faq.bandprotocol.com/
```

Reply:
Well the relay structure will be the same, we will only need new functions that accepts those kind of data rather than `RefData`structs